### ISSUE TITLE
About mist_layer code

### ISSUE THREAD

**Issue by WangBoying** [NVlabs/wetectron#18]
Sorry, I don’t understand this part of your code very well.
Could you please explain to me？ 
Thank you!
           **_boxes = proposals.bbox[max_inds.t().contiguous().view(-1)].view(num_gt_cls.int(), -1, 4)
            _boxes = BatchBoxList(_boxes, proposals.size, mode=proposals.mode)
            ious = batch_boxlist_iou(_boxes, _boxes)
            k_ind = torch.zeros(num_gt_cls.int(), k, dtype=torch.bool, device=device)
            k_ind[:, 0] = 1 # always take the one with max score 
            for ii in range(1, k):
                max_iou, _ = torch.max(ious[:,ii:ii+1, :ii], dim=2)
                k_ind[:, ii] = (max_iou < self.iou_th).byte().squeeze(-1)**

**Comment by jason718** [comment]
this code samples proposal boxes based on max_inds and the convert the samples into the BatchBoxList structure for computing IoUs between boxes in a batch manner. We then compute k_ind by (1) always take the 1st one (2) if proposal ii not overlap (<iou_th) with any proposal before it, we add it to k_ind.
